libs/positions/src/lib/close-position-dialog/complete.tsx

```diff
@@ -0,0 +1,108 @@
+import { useEnvironment } from '@vegaprotocol/environment';
+import type { OrderEvent_busEvents_event_Order } from '@vegaprotocol/orders';
+import { t, truncateByChars } from '@vegaprotocol/react-helpers';
+import { OrderRejectionReasonMapping, OrderStatus } from '@vegaprotocol/types';
+import { Link } from '@vegaprotocol/ui-toolkit';
+import type { TransactionResult, VegaTxState } from '@vegaprotocol/wallet';
+import type { ClosingOrder as IClosingOrder } from '../use-close-position';
+import { useRequestClosePositionData } from '../use-request-close-position-data';
+import { ClosingOrder } from './shared';
+
+interface CompleteProps {
+  partyId: string;
+  transaction: VegaTxState;
+  transactionResult?: TransactionResult;
+  closingOrder?: IClosingOrder;
+  closingOrderResult?: OrderEvent_busEvents_event_Order;
+}
+
+export const Complete = ({
+  partyId,
+  transaction,
+  transactionResult,
+  closingOrder,
+  closingOrderResult,
+}: CompleteProps) => {
+  const { VEGA_EXPLORER_URL } = useEnvironment();
+
+  if (!transactionResult || !closingOrderResult) return null;
+
+  return (
+    <>
+      {closingOrderResult.status === OrderStatus.STATUS_FILLED &&
+      transactionResult.status ? (
+        <Success partyId={partyId} order={closingOrder} />
+      ) : (
+        <Error
+          transactionResult={transactionResult}
+          closingOrderResult={closingOrderResult}
+        />
+      )}
+      {transaction.txHash && (
+        <>
+          <p className="font-semibold mt-4">{t('Transaction')}</p>
+          <p>
+            <Link
+              href={`${VEGA_EXPLORER_URL}/txs/${transaction.txHash}`}
+              target="_blank"
+            >
+              {truncateByChars(transaction.txHash)}
+            </Link>
+          </p>
+        </>
+      )}
+    </>
+  );
+};
+
+const Success = ({
+  partyId,
+  order,
+}: {
+  partyId: string;
+  order?: IClosingOrder;
+}) => {
+  const { market, marketData, orders } = useRequestClosePositionData(
+    order?.marketId,
+    partyId
+  );
+
+  if (!market || !marketData || !orders) {
+    return <div>{t('Loading...')}</div>;
+  }
+
+  if (!order) {
+    return (
+      <div className="text-vega-red">{t('Could retrieve closing order')}</div>
+    );
+  }
+  return (
+    <>
+      <h2 className="font-bold">{t('Position closed')}</h2>
+      <ClosingOrder order={order} market={market} marketData={marketData} />
+    </>
+  );
+};
+
+const Error = ({
+  transactionResult,
+  closingOrderResult,
+}: {
+  transactionResult: TransactionResult;
+  closingOrderResult: OrderEvent_busEvents_event_Order;
+}) => {
+  const reason =
+    closingOrderResult.rejectionReason &&
+    OrderRejectionReasonMapping[closingOrderResult.rejectionReason];
+  return (
+    <div className="text-vega-red">
+      {reason ? (
+        <p>{reason}</p>
+      ) : (
+        <p>
+          {t('Something went wrong')}: {transactionResult.error}
```

Comment:
(non-blocking): Why do we say `Something went wrong` instead of `Transaction failed because ...` or `Transaction failed. Error: ...`
It gives the impression that the environment is not stable like we don't know what went wrong. If we have an error we should know what went wrong.

Reply:
I agree. I'll update